### PR TITLE
INSP: Auto import quick fix

### LINKED ISSUE #1366: Autoimport of missing use
Is it possible to have autoimport of missing 'use' statements with Alt-Enter?

Right now it gives the suggestion in the error message but doesn't seem to have the auto fix option.

### LINKED ISSUE #2034: Highlight unknown identifiers and suggest quick-fix tips with relevant and applicable imports
I am coming from java/scala world, where IDE does this: if an identifier is unknown, it highlights it and suggest which definition one to import (if any).

Rust plugin shows this (notice `HttpConnector` variable is not known in the scope):

image

I would expect the plugin to highlight this variable by red-line underscores and offer a tip to `use hyper::client::HttpConnector` automatically.

VS Code + RLS does similar thing:
image
but I would expect one more tip in the quick-fix list: use fully qualified path to a variable instead of importing via use statement.

### PR DESCRIPTION
**Pull request by Undin**
Finally, it works :tada:
screen shot 2018-03-01 at 03 12 52

* convert auto import intention into quick fix for `RsUnresolvedReferenceInspection`
* enable by default `RsUnresolvedReferenceInspection`
* add option to ignore unresolved references without quick fix

Closes #1366
Closes #2034

### CODE REVIEW

**Comment by Undin** [comment]
@matklad or @vlad20012 r?

File: src/main/kotlin/org/rust/ide/inspections/fixes/import/AutoImportFix.kt [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+import org.rust.lang.core.psi.ext.*
+import org.rust.lang.core.stubs.index.RsNamedElementIndex
+import org.rust.lang.core.stubs.index.RsReexportIndex
+import org.rust.lang.core.types.ty.TyPrimitive
+import org.rust.openapiext.Testmark
+import org.rust.openapiext.runWriteCommandAction
+
+class AutoImportFix(path: RsPath) : LocalQuickFixOnPsiElement(path), HighPriorityAction {
+
+    private var isConsumed: Boolean = false
```
[the remark this answers is not in the document]
Comment (Undin) [reply src/main/kotlin/org/rust/ide/inspections/fixes/import/AutoImportFix.kt]: > This bit of mutable state looks suspicious...

I don't like it too but without it, we can get the following situation after successful auto import application
screen shot 2018-03-01 at 11 16 23
I don't know if it is bug or feature but looks like that platform tries to check availability of `HintAction` after its application

> Is this pattern used in other extensions?

`com.jetbrains.python.codeInsight.imports.AutoImportQuickFix` uses same approach

**Comment by Undin** [comment]
@matklad other comments?

**Comment by matklad** [comment]
@Undin I don't have time for a thorough review at the moment, but all in all it looks great, and I can't wait to try the nightly build with this feature, so, let's r+ and ask @vlad20012 to scrutinize this after the merge? :-)

**Comment by Undin** [comment]
bors r=matklad

**Comment by Undin** [comment]
bors r=matklad

**Comment by Undin** [comment]
bors r=matklad

**Pull request merged**